kart/cli.py

```diff
@@ -233,6 +246,210 @@ def push(ctx, do_progress, args):
+@click.option(
+    "--socket",
+    "socket_filename",
+    default=Path.home() / ".kart.socket",
+    show_default=True,
+    help="What socket to use",
+)
+@click.option(
+    "--timeout",
+    "timeout",
+    default=300,
+    show_default=True,
+    help="Timeout and shutdown helper when no commands received with this time",
+)
+@click.argument("args", nargs=-1, type=click.UNPROCESSED)
+def helper(ctx, socket_filename, timeout, args):
+    """Start the background helper process to speed up interaction"""
+    if is_windows:
+        click.echo("Helper mode not currently supported on Windows")
+        ctx.exit(1)
+
+    load_commands_from_args(["--help"])
+
+    # stash the required environment for used libs
+    # TODO - this should be checked to ensure it is all that is needed
+    #  is there anything beyond these which is needed, eg. PWD, USER, etc.
+    required_environment = {
+        k: v
+        for k, v in os.environ.items()
+        if k
+        in [
+            "PATH",
+            "LD_LIBRARY_PATH",
+            "LD_LIBRARY_PATH_ORIG",
+            "SPATIALINDEX_C_LIBRARY",
+            "GIT_CONFIG_NOSYSTEM",
+            "GIT_EXEC_PATH",
+            "GIT_TEMPLATE_DIR",
+            "GIT_INDEX_FILE",
+            "GDAL_DATA",
+            "PROJ_LIB",
+            "PROJ_NETWORK",
+            "OGR_SQLITE_PRAGMA",
+            "XDG_CONFIG_HOME",
+            "GIT_EXEC_PATH",
+            "GIT_TEMPLATE_DIR",
+            "GIT_INDEX_FILE",
+            "SSL_CERT_FILE",
+        ]
+    }
+
+    sock = socket.socket(family=socket.AF_UNIX)
+    os.umask(0o077)
+    try:
+        # TODO - this will take over a socket from an existing running helper process
+        #  it should check to see if another helper is already running and bail out
+        if os.path.exists(socket_filename):
+            os.unlink(socket_filename)
+
+        sock.bind(str(socket_filename))
+    except OSError as e:
+        print(dir(e))
+        click.echo(f"Unable to bind to socket [{socket_filename}] [{e.strerror}]")
+        ctx.exit(1)
+
+    sock.listen()
+
+    # ignore SIGCHLD so zombies don't remain when the child is complete
+    signal.signal(signal.SIGCHLD, signal.SIG_IGN)
+
+    # import modules that are deferred loaded in normal kart execution
+    from .tabular.working_copy.gpkg import WorkingCopy_GPKG
+    import kart
+    import osgeo
+    import psycopg2
+    import pysqlite3
+    import rtree
+
+    import sqlalchemy
+    from kart.sqlalchemy.gpkg import Db_GPKG
+    import pygments.token as token
+    from pygments.formatters import TerminalFormatter
+    import pygments.styles.default
+
+    while True:
+        # The helper will exit if no command received within timeout
+        sock.settimeout(timeout)
+        try:
+            client, info = sock.accept()
+            if os.fork() == 0:
+                payload, fds = recv_fds(client, 8164, 4)
+                # print("kart helper: handlng request...", payload)
```

Comment:
No need to check in commented out debug-code

Reply:
Yep.